api/v1alpha1/modelregistry_types.go

```diff
@@ -335,6 +335,18 @@ type OAuthProxyConfig struct {
 	// Create an OpenShift Route for REST proxy Service, enabled by default
 	ServiceRoute string `json:"serviceRoute,omitempty"`
 
+	// Domain name for Route configuration.
+	// If not provided, it is set automatically using model registry operator env variable DEFAULT_DOMAIN.
+	// If the env variable is not set, it is set to the OpenShift `cluster` ingress domain in an OpenShift cluster.
+	Domain string `json:"domain,omitempty"`
```

Comment:
What about adding additional validation as this string would be added to the .spec.host of the Openshift Route?

```yaml
host:
    description: |-
      host is an alias/DNS that points to the service. Optional.
      If not specified a route name will typically be automatically
      chosen.
      Must follow DNS952 subdomain conventions.
    maxLength: 253
    pattern: ^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$
    type: string
```

Reply:
Good suggestion @Al-Pragliola , adding it now.